optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/impl/domain/solution/JacksonSolutionFileIO.java

```diff
@@ -0,0 +1,57 @@
+package org.optaplanner.persistence.jackson.impl.domain.solution;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.persistence.common.api.domain.solution.SolutionFileIO;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ */
+public class JacksonSolutionFileIO<Solution_> implements SolutionFileIO<Solution_> {
+
+    protected Class<Solution_> clazz;
+    protected ObjectMapper mapper;
```

Comment:
Please make the class `final`, the fields `private final`.

Reply:
Sure thing i could convert the fields to be `private final`. Since the `clazz` and the `mapper` should never be changed after initialization. 

However, wouldn't making the class `final` prevent clients from extending this SolutionFileIO implementation for there own planning solution.

Therefore, clients would not be able to use the SolutionFileIO as explained in the documentation similar to XStreamSolutionFIleIO.

I made fields `private final` and waiting for your feedback on making the class `final`.

Reply:
Good point on the class. I forgot we would be extending it. Nevermind then.

Reply:
Great! Should I mark as resolved?

Reply:
Resolving.